llmfoundry/models/layers/attention.py

```diff
@@ -256,8 +256,9 @@ def flash_attn_fn(
 
         past_key_value = (key, value)
 
-    if attn_bias is not None:
-        raise NotImplementedError(f'attn_bias not implemented for flash attn.')
+    if attn_bias is not None and not is_flash_v2_installed(v2_version='v2.4.2'):
```

Comment:
Does this still work if we update the fa2 version? i.e. If you have a version higher than 2.4.2, will `is_flash_v2_installed(v2_version='v2.4.2')` return False?

Reply:
Yes, this will still work `is_flash_v2_installed` checks if at least the `v2_version` provided is installed:

https://github.com/mosaicml/llm-foundry/blob/083b4b2451a60487f0ff891b4fbd671064b90a30/llmfoundry/models/layers/attention.py#L27C16-L27C16